src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -0,0 +1,149 @@
+        this.timelineSync = timelineSync;
+        this.nakadiSettings = nakadiSettings;
+        this.timelineDbRepository = timelineDbRepository;
+        this.topicRepositoryHolder = topicRepositoryHolder;
+    }
+
+    public void createTimeline(final TimelineRequest timelineRequest, final Client client)
+            throws ForbiddenAccessException, TimelineException, TopicRepositoryException {
+        if (!client.getClientId().equals(securitySettings.getAdminClientId())) {
+            throw new ForbiddenAccessException("Request is forbidden for user " + client.getClientId());
+        }
+
+        try {
+            final String eventTypeName = timelineRequest.getEventType();
+            final EventType eventType = eventTypeCache.getEventType(eventTypeName);
+
+            final Storage storage = storageDbRepository.getStorage(timelineRequest.getStorageId())
+                    .orElseThrow(() -> new NotFoundException("No storage with id: " + timelineRequest.getStorageId()));
+            final Timeline activeTimeline = getTimeline(eventType);
+            final TopicRepository currentTopicRepo =
+                    topicRepositoryHolder.getTopicRepository(activeTimeline.getStorage());
+            final TopicRepository nextTopicRepo = topicRepositoryHolder.getTopicRepository(storage);
+            final List<PartitionStatistics> partitionStatistics =
+                    currentTopicRepo.loadTopicStatistics(Collections.singleton(activeTimeline.getTopic()));
+
+            if (activeTimeline.isFake()) {
+                LOG.info("Switching from fake timeline to real one. Current topic {}", activeTimeline.getTopic());
+                final Timeline nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, activeTimeline.getTopic(), new Date());
+                switchTimeline(eventType, nextTimeline, () -> nextTimeline.setSwitchedAt(new Date()));
+            } else {
+                final String newTopic = nextTopicRepo.createTopic(partitionStatistics.size(),
+                        eventType.getOptions().getRetentionTime());
+                final Timeline nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, newTopic, new Date());
+                LOG.info("Switching timelines. Topics: {} -> {}", activeTimeline.getTopic(), nextTimeline.getTopic());
+                switchTimeline(eventType, nextTimeline, () -> {
+                    final Timeline.StoragePosition storagePosition =
+                            StoragePositionFactory.createStoragePosition(activeTimeline, currentTopicRepo);
+                    activeTimeline.setLatestPosition(storagePosition);
+                    nextTimeline.setSwitchedAt(new Date());
+                });
+            }
+        } catch (final NakadiException ne) {
+            throw new TimelineException(ne.getMessage(), ne);
```

Comment:
It's strange that you wrap all Nakadi exceptions to TimelineException.
As I see from the code TimelineException will result in 503 response. But you can get here exceptions because event-type is not found or storage is not found - these all should be 422, not 503. That's why I don't understand this wrapping.